### PR TITLE
Fixes for unit tests for CoreLibrary

### PR DESCRIPTION
**Pull request by edleno2**
## Description

- Added new exception type FormatException so we can return same exception type as .net Core for Parse errors.
- Upgraded tests to check for exception assertion of type FormatException.
- Fixed a few exception assertions to be ArgumentOutOfRangeException to be compliant with .net core.
- Cleaned up unit tests, removed debug statements for tests that have now been fixed, added messages to some Asserts to make locating the failing tests more easy.
- Upgraded version to go with new nanoCLR checksum.
- Added more tests for invalid Parse strings for integers and floating point numbers.
- 

## Motivation and Context

- Fixes nanoFramework/Home#708

## How Has This Been Tested?

Run with Win32 - with and without SUPPORT_ANY_BASE_CONVERSION defined.  Run on ESP32-WROOM and STM32F429I-Discovery.
## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [x] Unit Tests (work on Unit Tests, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: nanoFramework.CoreLibrary/CoreLibrary.nfproj [hunk 1]
```diff
@@ -107,6 +107,7 @@
     <Compile Include="System\EventHandler.cs" />
     <Compile Include="System\Exception.cs" />
     <Compile Include="System\FlagsAttribute.cs" />
+    <Compile Include="System\FormatException.cs" />
```
Comment (josesimoes) [inline nanoFramework.CoreLibrary/CoreLibrary.nfproj:110]: You need to add this file to the non-reflection project too (as linked file).
This requires regenerating the assembly declaration for it and update nf-interpreter.

**Comment by josesimoes** [comment]
Nice improvement!! 💯 👍🏻

File: nanoFramework.CoreLibrary/CoreLibrary.nfproj (hunk 1 above)
Comment (edleno2) [reply nanoFramework.CoreLibrary/CoreLibrary.nfproj:110]: Ahhh...that's why the MDP didn't generate a changed stub!  Will do!

Comment (edleno2) [reply nanoFramework.CoreLibrary/CoreLibrary.nfproj:110]: Now added to no-reflection project and new checksums added to mscorlib in nanoCLR.

**Review by josesimoes — APPROVED** [review]
LGTM!

File: nanoFramework.CoreLibrary/CoreLibrary.nfproj (hunk 1 above)
Comment (josesimoes) [reply nanoFramework.CoreLibrary/CoreLibrary.nfproj:110]: Nice. I've fixed the inclusion so it's linked like all the rest. 

Also added back the System folder.

**Pull request merged**